VirtoCommerce.CatalogModule.Web/Controllers/Api/CatalogModuleListEntryController.cs

```diff
@@ -116,6 +120,66 @@ public IHttpActionResult CreateLinks(webModel.ListEntryLink[] links)
             return StatusCode(HttpStatusCode.NoContent);
         }
 
+        /// <summary>
+        /// Bulk create links to categories and items
+        /// </summary>
+        /// <param name="createLinksModel"></param>
+        /// <returns></returns>
+        [HttpPost]
+        [Route("~/api/catalog/listentrylinks/bulkcreate")]
+        [ResponseType(typeof(void))]
+        public IHttpActionResult BulkCreationLinks(webModel.BulkCreateLinkParameters createLinksModel)
+        {
+            var coreModelCriteria = createLinksModel.SearchCriteria.ToCoreModel();
+
+            bool haveProductsOrCategories;
+
+            do
+            {
+                var links = new List<webModel.ListEntryLink>();
+
+                var searchResult = _searchService.Search(coreModelCriteria);
+
+                var productLinks = searchResult
+                    .Products
+                    .Select(x => new webModel.ListEntryLink
+                    {
+                        CatalogId = createLinksModel.CatalogId,
+                        ListEntryType = webModel.ListEntryProduct.TypeName,
+                        ListEntryId = x.Id,
+                        CategoryId = createLinksModel.CategoryId
+                    })
+                    .ToList();
+
+                links.AddRange(productLinks);
+
+                var categoryLinks = searchResult
```

Comment:
And as I mentioned `searchResult.Categories` doesn't supports pagination for Categories and can be moved out from do while loop.